### PR TITLE
More `out` args

### PR DESCRIPTION
**Pull request by vidartf**
Fixed missing `out` arg in overloaded sum function in `EDSSpectrum`, and added `out` arg to several methods:
- `as_spectrum()`
- `as_image()`
- `rebin()`
- `diff()`
- `derivative()`
- `integrate_simpson()`
- `integrate1D()`
- `valuemax()`
- `get_histogram()`

### CODE REVIEW

**Comment by vidartf** [comment]
Merge in master and 0.8.x for a nice diff :)

**Comment by vidartf** [comment]
Possible further candidates I've been able to identify include methods on `Model`/`Component` (`as_signal` ?), and `Signal1DTools.integrate_in_range()`. Thoughts?

**Comment by francisco-dlp** [comment]
The have merged the current master into my branch but the diff is far from nice, what's going on?

**Comment by vidartf** [comment]
I'll rebase onto your branch, give me a sec.

File: hyperspy/signal.py [hunk 1]
```diff
@@ -4051,6 +4066,10 @@ def min(self, axis, out=None):
         axis : {int | string}
            The axis can be specified using the index of the axis in
            `axes_manager` or the axis name.
+        out : {Signal, None}
```
Comment (francisco-dlp) [inline hyperspy/signal.py]: Wouldn't it be better to add the `out` documentation to `hyperspy/doctrings`. See [e.g. here](https://github.com/hyperspy/hyperspy/blob/master/hyperspy/__init__.py#L31).

**Comment by vidartf** [comment]
Rebased with nice and tidy diff :)

File: hyperspy/signal.py (hunk 1 above)
Comment (vidartf) [inline hyperspy/signal.py]: Sure, I was thinking about this myself, but I figured I'd see what you though :)

**Comment by francisco-dlp** [comment]
Except for the docstring thing, this looks fine to me.

**Comment by vidartf** [comment]
I've extracted the docstring, but I was a little uncertain where to put it. I figured that since docstrings was a sub-package that I should add it to a new file `hyperspy/docstrings/signal.py`. If the thought was to keep them all in one file, then why is not simply a module `hyperspy/docstrings.py`?

**Comment by francisco-dlp** [comment]
`hyperspy/docstrings/signal.py` seems like a good location to me.

**Pull request merged**